openwisp_firmware_upgrader/private_storage/views.py

```diff
@@ -0,0 +1,22 @@
+from openwisp_firmware_upgrader.models import FirmwareImage
+from private_storage.views import PrivateStorageDetailView
+
+
+class FirmwareImageDownloadView(PrivateStorageDetailView):
+    model = FirmwareImage
+    model_file_field = 'file'
+
+    slug_field = 'file'
+    slug_url_kwarg = 'file'
+
+    def can_access_file(self, private_file):
+        user = private_file.request.user
+        if not user.is_authenticated:
+            return False
+        image = FirmwareImage.objects.get(file=private_file.relative_name)
```

Comment:
if you try to access use the `get_object` method of `PrivateStorageDetailView`, what comes out?

Reply:
Used self.object instead :+1: